### ISSUE TITLE
Add in Work Party Reminder Email Templates

### ISSUE THREAD

**Issue by drewpiston** [washingtontrails/vms#424]
Julie pointed us to a folder with email reminder templates (folders B and C):
https://drive.google.com/folderview?id=0B9Z_h99_tSEyLUxiVC0yc2lBUW8&usp=sharing

We can add these, but unfortunately they take a little longer than normal templates because they are VF (we need the VF templates in order to pull in the Required Doc stuff).

**Comment by drewpiston** [comment]
Hi Eliot, assigning to you to prioritize. Thanks!

**Comment by ebskopin** [comment]
Thanks Drew.  These would be important to have.

**Comment by drewpiston** [comment]
Thanks @ebskopin -- follow-up question:
Originally we implemented functionality for allowing different reminder templates based on recipient type as well. In other words, a Group volunteer would get a slightly different 5 week reminder email for a BCRT Work Party than a My Backpack volunteer for that same Work Party. I haven't seen email templates at that level of granularity -- is that something we still need?

**Comment by mscholtz** [comment]
@ebskopin @drewpiston where do we stand with this one?

**Comment by ebskopin** [comment]
Sorry for the delay.  We haven't put together templates with that specificity and won't be needing that at this point. The templates in that folder are important to have.

**Comment by mscholtz** [comment]
Thanks, @ebskopin.
@drewpiston given what Elliot said, can you give me a quick sense of what's left to do here at this point?

**Comment by drewpiston** [comment]
@mscholtz Yes, the VF email templates for reminder emails still need to be created based on Julie's folder (with correct dynamic links). Shouldn't take horribly long to complete, especially since they don't need the recipient level of granularity.  

I'll push forward with this one.

**Comment by drewpiston** [comment]
@ebskopin ,

Testing this one might be a little weird -- this is how it is setup:

http://screencast.com/t/BpkQO0eAY

Basically, on the Work Party Type the Email Template Ids need to be written into all of those fields (since we were expecting more granularity) and the Days Prior needs to be set as well.

Let me know if you'd like more help with testing.

 Drew

**Comment by ebskopin** [comment]
I've tried to test this and am not having any luck. One question, does the WP need to be in the "Published" state for the reminder emails to issue? @drewpiston

**Comment by drewpiston** [comment]
Hi @ebskopin ,

That one is dependent on a scheduled job, which evidently was set to expire sometime last year. I re-scheduled the job, so you should be able to test now.

Cheers,
 Drew

**Comment by mscholtz** [comment]
Drew, which class are we talking about here?  I want to double check that
it's on my list to schedule in production.

On Fri, Jan 8, 2016 at 4:08 PM, drewpiston <notifications@github.com> wrote:

> Hi @ebskopin <https://github.com/ebskopin> ,
>
> That one is dependent on a scheduled job, which evidently was set to
> expire sometime last year. I re-scheduled the job, so you should be able to
> test now.
>
> Cheers,
> Drew
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/424#issuecomment-170161880>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by drewpiston** [comment]
Class is WTA_VMS_WP_Email_Alert_Batch.



--
www.drewpiston.com

**Comment by drewpiston** [comment]
Hey @ebskopin ,

I think my testing instructions were wrong/unclear. It looks like we have some old fields on the WP Type (e.g., Send 1st Reminder Email, Days Prior) that are no longer used. Instead, to set this up, we need to actually create "Email Reminder" records as children of the WP Type, like this one:
https://cs9.salesforce.com/a30K000000022MM

@mscholtz , I think we should remove those old fields from the layout, or maybe even delete them. Do you agree? I'll see if I can find anything written down where we decided to abandon them.

**Comment by mscholtz** [comment]
If you can verify that they're no longer being used, then yes, we should
delete them.  If you do, can you send me a list of fieldnames you deleted
so I can do the same in production?

On Thu, Jan 14, 2016 at 3:38 PM, drewpiston <notifications@github.com>
wrote:

> Hey @ebskopin <https://github.com/ebskopin> ,
>
> I think my testing instructions were wrong/unclear. It looks like we have
> some old fields on the WP Type (e.g., Send 1st Reminder Email, Days Prior)
> that are no longer used. Instead, to set this up, we need to actually
> create "Email Reminder" records as children of the WP Type, like this one:
> https://cs9.salesforce.com/a30K000000022MM
>
> @mscholtz <https://github.com/mscholtz> , I think we should remove those
> old fields from the layout, or maybe even delete them. Do you agree? I'll
> see if I can find anything written down where we decided to abandon them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/424#issuecomment-171817836>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by drewpiston** [comment]
Ok, I deleted:
Send_1st_Reminder_Email_Days_Prior__c
Send_2nd_Reminder_Email_Days_Prior__c

I'll push to git, too.

**Comment by mscholtz** [comment]
Got it, thanks.

On Tue, Jan 19, 2016 at 3:34 PM, drewpiston <notifications@github.com>
wrote:

> Ok, I deleted:
> Send_1st_Reminder_Email_Days_Prior__c
> Send_2nd_Reminder_Email_Days_Prior__c
>
> I'll push to git, too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/424#issuecomment-173023271>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/
